### ISSUE TITLE
Pdf viewer issues with Atom 1.39

### ISSUE THREAD

**Issue by ashthespy** [James-Yu/Atom-LaTeX#206]
The new upgrade from Electron `2.x` to `3.x` (https://github.com/atom/atom/pull/19419) seems to have broken the pdf viewer. 
#### Tab viewer  - freezes after a few minutes 
I was initially under the impression that it could be caused by the `webview` the tab viewer uses, but switching to `iframe` doesn't seem to help either.
```diff
index e8d0e42..9bd7367 100644
--- a/lib/viewer.coffee
+++ b/lib/viewer.coffee
@@ -126,7 +126,10 @@ class Viewer extends Disposable
 
 class PDFView
   constructor: (url,title) ->
-    @element = document.createElement 'webview'
+    @element = document.createElement 'iframe'
+    @element.setAttribute 'width', '100%'
+    @element.setAttribute 'height', '100%'
+    @element.setAttribute 'frameborder', 0,
     @element.setAttribute 'src', url
     @element.addEventListener 'console-message', (e) -> console.log e.message
     @title = title
```
#### Window viewer  - closes after a few minutes 
Anyone else seeing these issues as well?

**Comment by Elijahh177** [comment]
I have the same issue.

**Comment by dpo** [comment]
Since the upgrade, the external viewer has been crashing and closing after varying amounts of time.

**Comment by ashthespy** [comment]
This seems to be an Atom/electron upgrade issue. With Atom 1.39, I tried downgrading the package (`apm install atom-latex@0.8.5`) but it doesn't seem to help at all. 
I tried down all the way down to `0.8.1`  to rule out different probable causes (webGL, webview, etc) but all of them seem to crash.  Unfortunately I am not well versed with the internals of electron to debug more. Until then, I am back to Atom 1.38

**Comment by dpo** [comment]
I also reverted back to Atom 1.38, which works well *and* #192 works!

**Comment by ashthespy** [comment]
I believe I found the issue - the server was hitting the default timeout and killing all connections, this making the websocket close, and thus the window close\tab freeze.  

Could someone else also confirm if the following fixes it for them? Will push a fix then for it.

```diff
diff --git a/lib/server.coffee b/lib/server.coffee
index 8f3069e..6d003d6 100644
--- a/lib/server.coffee
+++ b/lib/server.coffee
@@ -17,15 +17,20 @@ class Server extends Disposable
           e(err)
         else if @latex.server.openTab
           @latex.viewer.openViewerNewWindow()
+    @http.setTimeout 1000, (soc) => {}
+    # @http.on "timeout", () => console.log 'Server:Timeout'
 
     @ws = new ws.Server server: @http
     @ws.on "connection", (ws) =>
       ws.on "message", (msg) => @latex.viewer.wsHandler(ws, msg)
-      ws.on "close", () => @latex.viewer.wsHandler(ws, '{"type":"close"}')
+      ws.on "close", () => 
+        console.timeEnd('ServerClosed')
+        @latex.viewer.wsHandler(ws, '{"type":"close"}')
       ws.on "error", (e) => console.error(e)
 
   httpHandler: (request, response) ->
     if request.url.indexOf('viewer.html') > -1
+      console.time('ServerClosed')
       response.writeHead 200, 'Content-Type': 'text/html'
       response.end fs.readFileSync("""#{@httpRoot}/viewer.html"""), 'utf-8'
       return
```

**Comment by dixr** [comment]
I can confirm that your patch fixed the issue for me.

**Comment by ashthespy** [comment]
Thanks for confirming, fixed with `v0.8.8`
